### PR TITLE
Add filesystem syscalls

### PR DESCRIPTION
**Pull request by pommicket**
Sorry for putting all this in one PR ! It ended up being quite large…
This PR adds the following system calls to KidneyOS:

- open
- read
- write
- lseek64
- close
- chdir
- getcwd
- mkdir
- rmdir
- fstat
- unlink
- getdents (system call version of readdir)
- link
- symlink
- rename
- ftruncate
- unmount
- mount (currently only supports tmpfs)
- sync

### CODE REVIEW

**Review by LanceLi1416 — APPROVED** [review]
Apologies for the delay in reviewing this PR. This is truly amazing work! Awaiting endorsement from the process team and the resolution of the merge conflict with `main`. Otherwise, everything looks great to me!

File: kernel/src/fs/mod.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+pub type FileDescriptor = i16;
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ProcessFileDescriptor {
+    pub pid: Pid,
+    pub fd: FileDescriptor,
+}
+
+unsafe fn running_process() -> &'static ProcessControlBlock {
```
Comment (mckeenicholas) [inline kernel/src/fs/mod.rs:17]: These would be useful as standalone functions within the system crate

Comment (pommicket) [reply kernel/src/fs/mod.rs:17]: Sure, I'll move them there

**Comment by pommicket** [comment]
I'll merge this now so I don't have to keep dealing with merge conflicts… But let me know if you have any more suggestions :)

**Pull request merged**